### ISSUE TITLE
It seems hyperHTML updating nodes unnecessarily?

### ISSUE THREAD

**Issue by prabu-ramaraj** [WebReflection/hyperHTML#64]
I have been reviewing multiple UI libraries for my next project. I am very impressed with hyperHTML as it is leveraging already available platform features like Template literals.  I am playing around now and I have this quick question.  My apologies, if this is a dumb question (I am learning).

In the article test here - https://webreflection.github.io/hyperHTML/test/article.html

1.  Go to this article.html page
2.  In the console, call the update() method

`update(
    hyperHTML.bind(articleElement),
    {
      title: 'True story',
      magic: true,
      paragraphs: [
        {title: 'touching'},
        {title: 'incredible'},
        {title: 'doge'}
      ]
    }
  );`

3. Notice that h3 tag (which has title) is getting updated unnecessarily. state is not changed from the initial value. Then, why h3 tag is always getting updated.

Currently h3 tag looks like this
`<h3>${state.title}</h3>`

If I change this to
`<h3>some text, ${state.title}</h3>`
then title is not getting updated unnecessarily

Honestly, I don't know if this is issue or not. Just trying to understand all these better.  Thank you so much for this awesome library. Can't wait for the "adopt" and refactoring release.

**Comment by WebReflection** [comment]
Don't get me wrong, it's just a talk slide I've showed today that should help you understand (specially second and third point)

![screenshot from 2017-06-20 09-05-19](https://user-images.githubusercontent.com/85749/27458400-c1270680-57a0-11e7-84eb-8fb01ce9a68e.png)

I'm not sure the example is bad all together but if you write the follogin:
```html
<h3>${state.title}</h3>
```
you are addressing `any` possible content in that `h3` tag while if you write this instead:
```html
<h3> ${state.title} </h3>
```
You are addressing the `#text` node content and not `any` content.

The difference is that `#text` uses a text node and it's safe and sanitized for you while `any` content accepts:
  * any text that will be injected as html
  * any DOM node
  * an Array of html strings or DOM nodes
  * eventually other things on the backend or the native end

Since the effort to understand if any content is the same is too complex, specially for an interpolation that could bring literally any sort of value, there is no guarantee whatsoever that if you inject HTML that's not going to happen, specially because HTML can contain any node, meaning it could trash any previous content and create any new one so that changes would be **meant**.

Indeed, without spaces surrounding that interpolation, you could have:
```html
<h3>${'<iframe src="http://reload.me"></iframe>'}</h3>
```

as well as 
```html
<h3>${'Hello <strong>world</strong>'}</h3>
```

and as you can see both creates new nodes.

Do this again?
```html
<h3>${'Hello <strong>world</strong>'}</h3>
```

There you have a whole new `<strong>` element to inject 'cause the previous one is gone.


**TL;DR** do not ever put `#text` content inside tags if it's `#text` content that you want, and do not expect `any` content to filter updates from strings that could contain anything as HTML and there's no way this library will include an HTML parser.

I hope this explains your gotcha, and maybe I shuold just update that bad example.

**Comment by prabu-ramaraj** [comment]
Thank you so much. I understood this fully now. In fact, I did read this in your FAQ section. But, was confused a bit as GITHUB readme file didn't show the space around ${'text'} below. I just referred the raw file and I do see one space in front of ${'text'}.  Thanks again.

![image](https://user-images.githubusercontent.com/12944736/27460817-4365c31a-5784-11e7-83bd-ed2f9e8f2c00.png)
